protocols/imap/src/main/java/org/apache/james/imap/processor/base/SelectedMailboxImpl.java

```diff
@@ -92,16 +93,30 @@ public SelectedMailboxImpl(MailboxManager mailboxManager, EventBus eventBus, Ima
 
         mailboxId = messageManager.getId();
 
+
+        applicableFlags = new AtomicReference<>();
+
         registration = Mono.from(eventBus.register(this, new MailboxIdRegistrationKey(mailboxId)))
             .subscribeOn(Schedulers.elastic())
             .block();
 
-        applicableFlags = messageManager.getApplicableFlags(mailboxSession);
+        applicableFlags.accumulateAndGet(messageManager.getApplicableFlags(mailboxSession), this::mergeFlags);
         try (Stream<MessageUid> stream = messageManager.search(new SearchQuery(SearchQuery.all()), mailboxSession)) {
             uidMsnConverter.addAll(stream.collect(Guavate.toImmutableList()));
         }
     }
 
+    private Flags mergeFlags(Flags a, Flags b) {
+        if (a == null) {
```

Comment:
how it can be null?

Reply:
When not yet initialized, mainly. AtomicRef is a crappy class to work with, sorry.